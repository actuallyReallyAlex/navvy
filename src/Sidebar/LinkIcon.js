import React from 'react'
import PropTypes from 'prop-types'
import * as grommetIcons from 'grommet-icons'

const LinkIcon = ({ icon }) => {
  const Icon = grommetIcons[icon]
  return <Icon />
}

LinkIcon.propTypes = {
  icon: PropTypes.string.isRequired
}

export default LinkIcon
