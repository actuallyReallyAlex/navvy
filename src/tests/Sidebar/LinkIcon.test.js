import React from 'react'
import LinkIcon from '../../Sidebar/LinkIcon'
import { shallow } from 'enzyme'

const defaultProps = {
  icon: 'Cube'
}

describe('<LinkIcon />', () => {
  afterEach(() => {})

  test('Should render the <LinkIcon /> component.', () => {
    const component = shallow(<LinkIcon {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
