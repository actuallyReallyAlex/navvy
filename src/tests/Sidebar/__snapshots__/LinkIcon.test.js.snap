// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkIcon /> Should render the <LinkIcon /> component. 1`] = `ShallowWrapper {}`;
